File: manual-tests/test2-capacitor-java-manually-convert-to-kotlin/android/src/main/java/com/mycompany/plugins/example/CapacitorJavaToKotlinPlugin.kt
package com.mycompany.plugins.example

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "CapacitorJavaToKotlin")
class CapacitorJavaToKotlinPlugin : Plugin() {
    private val implementation = CapacitorJavaToKotlin()

    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value")

        val ret = JSObject()
        ret.put("value", implementation.echo(value))
        call.resolve(ret)
    }
}

File: manual-tests/test2-capacitor-java-manually-convert-to-kotlin/ios/Sources/CapacitorJavaToKotlinPlugin/CapacitorJavaToKotlinPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorJavaToKotlinPlugin)
public class CapacitorJavaToKotlinPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CapacitorJavaToKotlinPlugin"
    public let jsName = "CapacitorJavaToKotlin"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = CapacitorJavaToKotlin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
